zerver/views/custom_profile_fields.py

```diff
@@ -78,6 +78,12 @@ def is_default_external_field(field_type: int, field_data: ProfileFieldData) ->
     return True
 
 
+def updating_subtype_of_external_account(
+    existing_field_data: ProfileFieldData, requested_field_data: ProfileFieldData
+) -> bool:
+    return existing_field_data["subtype"] != requested_field_data["subtype"]
```

Comment:
This needs a block comment above it that explains __why__ we're doing this -- specifically that we allow changing the name and hint fields, but not the field type.

Also ... where are you validating that the URL pattern, or other fields, can't be changed via this API? We really only want to support modifying name and hint, right?

Reply:
>This needs a block comment above it that explains why we're doing this -- specifically that we allow changing the name and hint fields, but not the field type.

Done, I updated the PR.

>Also ... where are you validating that the URL pattern, or other fields, can't be changed via this API? We really only want to support modifying name and hint, right?

I shouldn't remove this line, I think I did mistake here
`if is_default_external_field(field.field_type, orjson.loads(field.field_data)):`
because the change from PR is only for the default external account type fields, and in its `field_data` only the `subtype` field is present.